### DISCUSSION TITLE
Bug: Warning: File ignored because of a matching ignore pattern. (git hook)

### DISCUSSION THREAD

**Discussion by infacto** [eslint/eslint#17437] [General]
### Environment

Node version: v18.17.0
npm version: v9.6.7
Local ESLint version: v8.24.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19045

### What parser are you using?

@typescript-eslint/parser

### What did you do?

I use husky with lint-stage for git hook on commit to execute eslint:

```json
{
  "lint-staged": {
    "*.{ts,js}": "eslint --fix --max-warnings=0"
  }
}
```

Max warnings zero because I use `only-warn` plugin for IDE to enable coding without errors which may stop compiling. 
The developer can see the warning and will fix it before committing it. So fail on commit only on open linter issues.

Now I want to commit an ignored file from project setup. But this is not possible because of a useless warning:

```
ESLint found too many warnings
0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override
```

### What did you expect to happen?

Why care about ignored files? Just proceed and don't throw a warning.

### What actually happened?

A warning appears because the passed file is marked to ignore.

### Link to Minimal Reproducible Example

Not possible or necessary in this case.

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Don't warn for ignored files. You could rather log on "info" level. But this is also just annoying. We could inform on the sumary log at the end that there are x ignore files. But I would just remove this.

I tried `--no-error-on-unmatched-pattern`, but this does not work.

**Comment by infacto** [comment]
Found another related issue: #15010

**Comment by fasttime** [comment]
Hi @infacto, thanks for the issue. This is not a bug: the warning says that a file you a trying to lint is being actually ignored, so I would say it's helpful. Since this is the same problem described in #15010 (you found it already), it would be best to continue the discussion there.

**Comment by infacto** [comment]
I'm not sure if a warning about linting ignored files is helpful. Generally, you just run the linter over the project, which doesn't handle ignored files anyway. If linter is running explicitly on an ignored file, an information is helpful, yes. Info not warning. Or at least provide an option to suppress it. I don't understand the discussion blocker. I see no base for discussion about this trivial option. It important especially for such (not rare) cases, and it is not really a critical feature. 

Forwarded to #15010 ...
